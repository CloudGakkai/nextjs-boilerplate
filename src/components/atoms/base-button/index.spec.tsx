import { render, screen } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

import userEvent from "@testing-library/user-event";

import { BaseButton } from ".";

describe("BaseButton", () => {
  it("should render the button with the label", () => {
    render(<BaseButton label="Primary" variant="primary" />);
    expect(screen.getByText("Primary")).toBeInTheDocument();
    expect(screen.getByText("Primary")).not.toBeDisabled();
  });

  it("should render the button with the label and disabled", () => {
    render(<BaseButton label="Primary" variant="primary" disabled />);
    expect(screen.getByText("Primary")).toBeInTheDocument();
    expect(screen.getByText("Primary")).toBeDisabled();
  });

  it("should render the button with the label and loading", () => {
    render(<BaseButton label="Primary" variant="primary" loading />);
    expect(screen.getByText("Primary")).toBeInTheDocument();
    expect(screen.getByText("Primary")).toBeDisabled();
  });

  it("should render the button with the label and click", async () => {
    const onClick = jest.fn();
    render(<BaseButton label="Primary" variant="primary" onClick={onClick} />);
    await userEvent.click(screen.getByText("Primary"));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});

describe("BaseButton Snapshots", () => {
  it("should render the button with the label and snapshot", () => {
    const { container } = render(
      <BaseButton label="Primary" variant="primary" />,
    );
    expect(container).toMatchSnapshot();
  });

  it("should render the button with the label and snapshot", () => {
    const { container } = render(
      <BaseButton label="Secondary" variant="secondary" />,
    );
    expect(container).toMatchSnapshot();
  });

  it("should render the button with the label and snapshot", () => {
    const { container } = render(
      <BaseButton label="Tertiary" variant="tertiary" />,
    );
    expect(container).toMatchSnapshot();
  });

  it("should render the button with the label and snapshot", () => {
    const { container } = render(<BaseButton label="Primary" />);
    expect(container).toMatchSnapshot();
  });
});
