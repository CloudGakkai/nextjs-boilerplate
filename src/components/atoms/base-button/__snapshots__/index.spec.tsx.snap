// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton Snapshots should render the button with the label and snapshot 1`] = `
<div>
  <div>
    <button
      class="bg-blue-500 text-white font-regular rounded-lg px-5 py-3 hover:opacity-80 disabled:opacity-50"
    >
      Primary
    </button>
  </div>
</div>
`;

exports[`BaseButton Snapshots should render the button with the label and snapshot 2`] = `
<div>
  <div>
    <button
      class="bg-white text-gray-600 font-regular rounded-lg px-5 py-3 hover:opacity-80 disabled:opacity-50"
    >
      Secondary
    </button>
  </div>
</div>
`;

exports[`BaseButton Snapshots should render the button with the label and snapshot 3`] = `
<div>
  <div>
    <button
      class="bg-red-500 text-white font-regular rounded-lg px-5 py-3 hover:opacity-80 disabled:opacity-50"
    >
      Tertiary
    </button>
  </div>
</div>
`;

exports[`BaseButton Snapshots should render the button with the label and snapshot 4`] = `
<div>
  <div>
    <button
      class="bg-blue-500 text-white font-regular rounded-lg px-5 py-3 hover:opacity-80 disabled:opacity-50"
    >
      Primary
    </button>
  </div>
</div>
`;
